lib/packages/docutils/rst.nim

```diff
@@ -912,6 +912,38 @@ template newLeaf(s: string): PRstNode = newRstLeaf(s)
 proc newLeaf(p: var RstParser): PRstNode =
   result = newLeaf(currentTok(p).symbol)
 
+proc validRefnamePunct(x: string): bool =
+  ## https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#reference-names
+  x.len == 1 and x[0] in {'-', '_', '.', ':', '+'}
+
+func getRefnameIdx(p: RstParser, startIdx: int): int =
```

Comment:
Should this be like `self: RstParser` according to the new style guide ?,
I know the code is older than the new guide, but maybe a follow up PR can improve it... :thinking:

Reply:
@juancarlospaco I don't think that we should strictly follow the style guide everywhere (regarding naming especially), and even if we want to, it should be done in a separate PR to not conflate with the actual PR changes.

Reply:
I agree; I literally said a separate PR.

Reply:
@juancarlospaco ,
my understanding is that using `self` is for object-oriented programming, and it's not quite the case for `rst.nim`: `p` contains tokens as input, and most (other) functions have argument `result: string` as their output. So it's more like procedural programming. And `p` is conventional for all procs in `rst.nim`.

BTW Nim Manual contains examples with keyword `using` for similar cases (https://nim-lang.org/docs/manual.html#statements-and-expressions-using-statement). So may be it's even worth to have `using p: RstParser` for this file and just type `p;` in the arguments.

Reply:
`self` shouldn't be part of our style guide... No compiler/stdlib code uses `self` anyway.